src/output/tests.rs

```diff
@@ -18,3 +18,31 @@ fn output_is_prettified() {
         pretty_body
     );
 }
+
+#[test]
+fn format_parsing_test() {
+    let result = parse_format("a%R,%H,%B");
+    assert_eq!(
+        result.expect("parse correctly"),
+        vec![
+            FormatItem::Chars("a".into()),
+            FormatItem::FirstLine,
+            FormatItem::Chars(",".into()),
+            FormatItem::Headers,
+            FormatItem::Chars(",".into()),
+            FormatItem::Body
+        ]
+    );
+
+    let result = parse_format("a%X");
+    assert!(result.is_err());
+    assert_eq!(
+        result.unwrap_err().to_string(),
+        "Invalid formatting character 'X'"
+    );
+    let result = parse_format("%R");
+    assert_eq!(
+        result.expect("parse correctly"),
+        vec![FormatItem::FirstLine,]
+    );
+}
```

Comment:
Could we add another test for verifying the behaviour we wanted with https://github.com/bayne/dot-http/issues/41? (only output body)